### PR TITLE
feat!: extract event component as a standalone piece

### PR DESCRIPTION
**Pull request by suhussai**
### Issue # (if applicable)

Closes #<issue number here>.

### Checklist

- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/awslabs/sbt-aws/blob/main/CONTRIBUTING.md)

---

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.*

### CODE REVIEW

**Review by ujwalbukka — APPROVED** [review]
Looks good to me !

File: src/control-plane/control-plane.ts [hunk 1]
```diff
@@ -2,65 +2,69 @@
...
-  readonly controlPlaneEventSource?: string;
+  readonly billing?: IBilling;
 
   /**
-   * The source to use for outgoing events that will be placed on the EventBus.
-   * This is used as the default if the OutgoingEventMetadata source field is not set.
+   * The event manager instance. If not provided, a new instance will be created.
    */
-  readonly applicationPlaneEventSource?: string;
+  readonly eventManager?: EventManager;
```
Comment (tobuck-aws) [inline src/control-plane/control-plane.ts:30]: Shouldn't this be IEventManager?

File: src/core-app-plane/core-app-plane.ts [hunk 2]
```diff
@@ -108,30 +71,12 @@ export interface CoreApplicationPlaneJobRunnerProps {
...
-   * This is used as the default if the IncomingEventMetadata source field is not set.
-   */
-  readonly controlPlaneEventSource?: string;
-
-  /**
-   * The source to use for outgoing events that will be placed on the EventBus.
-   * This is used as the default if the OutgoingEventMetadata source field is not set.
-   */
-  readonly applicationPlaneEventSource?: string;
+  readonly eventManager?: EventManager;
```
Comment (tobuck-aws) [inline src/core-app-plane/core-app-plane.ts:74]:
```suggestion
  readonly eventManager?: IEventManager;
```

File: src/core-app-plane/core-app-plane.ts [hunk 3]
```diff
@@ -141,22 +86,16 @@ export interface CoreApplicationPlaneProps {
  * and respond to events created by the control plane.
  */
 export class CoreApplicationPlane extends Construct {
+  /**
+   * The EventManager instance that allows connecting to events flowing between
+   * the Control Plane and other components.
+   */
   readonly eventManager: EventManager;
```
Comment (tobuck-aws) [inline src/core-app-plane/core-app-plane.ts:93]:
```suggestion
  readonly eventManager: IEventManager;
```

File: src/core-app-plane/core-app-plane.ts [hunk 4]
```diff
@@ -1,48 +1,11 @@
...
-  readonly detailType: string[];
-
-  /**
-   * The list of sources to listen for in the incoming event.
-   *
-   * @default CoreApplicationPlaneProps.controlPlaneEventSource
-   */
-  readonly source?: string[];
-}
+import { EventManager, DetailType, setTemplateDesc } from '../utils';
```
Comment (tobuck-aws) [inline src/core-app-plane/core-app-plane.ts:8]:
```suggestion
import { IEventManager, DetailType, setTemplateDesc } from '../utils';
```

**Review by tobuck-aws — CHANGES_REQUESTED** [review]
A few minor comments.

File: src/core-app-plane/core-app-plane.ts (hunk 4 above)
Comment (suhussai) [reply src/core-app-plane/core-app-plane.ts]: Needs both `EventManager` and `IEventManager`

**Review by tobuck-aws — APPROVED** [review]

**Pull request merged**